test/uk/gov/hmrc/pensionschemereturn/transformations/nonsipp/MemberPaymentsTransformerSpec.scala

```diff
@@ -32,5 +39,203 @@ class MemberPaymentsTransformerSpec extends EtmpTransformerSpec {
+          surrenderedBenefits = None
+        )
+
+        val etmpMemberPayments = transformEtmpSurrenderedBenefits(
+          surrenderMade = false,
+          surrenderedBenefits = None
+        )
+
+        val result = memberPaymentsTransformer.toEtmp(memberPayments)
+
+        result shouldMatchTo etmpMemberPayments
+      }
+    }
+
+    "successfully transform surrenderedBenefits section from ETMP format" when {
+      "surrenders made is TRUE, BOTH surrendered benefits are provided" in {
+
+        val etmpMemberPayments = transformEtmpSurrenderedBenefits(
+          surrenderMade = true,
+          surrenderedBenefits = Some(List(etmpPensionSurrender))
+        )
+
+        val memberPayments = transformSurrenderedBenefits(
+          made = true,
+          completed = true,
+          surrenderedBenefits = Some(pensionSurrender)
+        )
+
+        val result = memberPaymentsTransformer.fromEtmp(etmpMemberPayments)
+
+        result shouldMatchTo Right(memberPayments)
+      }
+
+      "surrenders made is TRUE, NO surrendered benefits are provided" in {
+
+        val etmpMemberPayments = transformEtmpSurrenderedBenefits(
+          surrenderMade = true,
+          surrenderedBenefits = Some(Nil)
+        )
+
+        val memberPayments = transformSurrenderedBenefits(
+          made = true,
+          completed = true,
+          surrenderedBenefits = None
+        )
+
+        val result = memberPaymentsTransformer.fromEtmp(etmpMemberPayments)
+
+        result shouldMatchTo Right(memberPayments)
+      }
+
+      "surrenders made is FALSE and BOTH surrendered benefits are provided" in {
+
+        val etmpMemberPayments = transformEtmpSurrenderedBenefits(
+          surrenderMade = false,
+          surrenderedBenefits = Some(List(etmpPensionSurrender))
+        )
+
+        val memberPayments = transformSurrenderedBenefits(
+          made = false,
+          completed = true,
+          surrenderedBenefits = Some(pensionSurrender)
+        )
+
+        val result = memberPaymentsTransformer.fromEtmp(etmpMemberPayments)
+
+        result shouldMatchTo Right(memberPayments)
+      }
+    }
   }
+
+  private def transformSurrenderedBenefits(
+    made: Boolean,
+    completed: Boolean,
+    surrenderedBenefits: Option[PensionSurrender]
+  ): MemberPayments =
+    sampleMemberPayments.copy(
+      benefitsSurrenderedDetails = SectionDetails(
+        made = made,
+        completed = completed
+      ),
+      memberDetails = sampleMemberPayments.memberDetails.map(_.copy(benefitsSurrendered = surrenderedBenefits))
+    )
+
+  private def transformEtmpSurrenderedBenefits(
+    surrenderMade: Boolean,
+    surrenderedBenefits: Option[List[EtmpPensionSurrender]]
+  ): EtmpMemberPayments =
+    sampleEtmpMemberPayments.copy(
+      surrenderMade = surrenderMade,
+      memberDetails = sampleEtmpMemberPayments.memberDetails.map(_.copy(memberPensionSurrender = surrenderedBenefits))
+    )
 }
+
+/**
+ *
+ * given 2 members
```

Comment:
Should we push this comment ?

Reply:
The logic is quite complicated so thought it would help but can get rid